libnavigation-core/src/main/java/com/mapbox/navigation/core/history/model/HistoryEventMapper.kt

```diff
@@ -105,6 +109,20 @@ internal class HistoryEventMapper {
         }
     }
 
+    private fun addWaypointNames(response: String): String {
```

Comment:
Why don't we run this function only if the `DirectionsResponse.fromJson(jsonResponse)` actually fails in the first place? Just to optimized it a little bit and avoid unnecessary work.

Reply:
Addressed the feedback. Let me know if it looks good to you - spoiler alert: I'm not convinced about the nested `try catch`s 😬